app/web_hub.go

```diff
@@ -101,7 +101,7 @@ func (a *App) HubStart() {
 				for _, hub := range a.Srv().GetHubs() {
 					if len(hub.broadcast) >= DEADLOCK_WARN {
 						mlog.Error(
-							"Hub processing might be deadlock with events in the buffer",
+							"Hub queue is almost full. It might be caused by a deadlock with events in the buffer",
```

Comment:
My point was that the hub queue can only be full when messages are being sent at a rate it is not able to process. And there is no deadlock anywhere in the code. But from https://github.com/mattermost/mattermost-server/pull/6100, it does seem like the codebase used to have deadlock conditions due to which processing would stop.

Is that still the case now @crspeller, @jwilander, @lieut-data ?

The context of this change is that, we started to see this message in our load tests when the queue was full just due to the sheer rate of messages. And I was surprised to see why the log message says there is a deadlock when there is none.

Reply:
Good question. I've left the deadlock message just in case but the main message is about having the queue almost full

Reply:
I believe you're correct @agnivade. I can't remember what the deadlock conditions were but I do remember we had to deal with it. Maybe @crspeller will remember more

Reply:
Ya this code definitely deadlocked years ago which is probably why this code exists. I don't remember the specifics. I do remember a certain large customer had a lot of fun with this deadlocking early on.

Reply:
Amazing, thanks for the context guys !

So this is a bit tricky. To be technically correct, the message should read - "Hub queue is full. This may be due to a deadlock, or just too many messages than the server can handle". But that's just equally confusing.

I think with the new metrics that @ethervoid has recently added, we have a much better insight into what's really going on with the hub. Additionally, I have plans to run extensive tests exclusively with this part of the codebase so that we become more confident when we make bigger architectural changes.

Given this, I think it may be simpler to just report what happened, rather than also suggest the cause for this, and just let the engineer debug it from the graphs and the stack trace. I will also love to get rid of this stack printing eventually, but that's for later.

Reply:
Good for me :)

Reply:
@ethervoid - Let's do the changes then ?